cmd/zoekt-sourcegraph-indexserver/main.go

```diff
@@ -843,22 +872,24 @@ func newServer(conf rootConfig) (*Server, error) {
 		debug = log.New(os.Stderr, "", log.LstdFlags)
 	}
 
-	indexingMetricsReposAllowlist := os.Getenv("INDEXING_METRICS_REPOS_ALLOWLIST")
-	if indexingMetricsReposAllowlist != "" {
+	reposWithSeparateIndexingMetrics = getEnvWithDefaultEmptySet("INDEXING_METRICS_REPOS_ALLOWLIST")
+	if len(reposWithSeparateIndexingMetrics) > 0 {
 		var repos []string
-
-		for _, r := range strings.Split(indexingMetricsReposAllowlist, ",") {
-			r = strings.TrimSpace(r)
-			if r != "" {
-				repos = append(repos, r)
-			}
+		for r := range reposWithSeparateIndexingMetrics {
+			repos = append(repos, r)
 		}
 
-		for _, r := range repos {
-			reposWithSeparateIndexingMetrics[r] = struct{}{}
+		debug.Printf("capturing separate indexing metrics for: %s", strings.Join(repos, ", "))
+	}
+
+	deltaBuildRepositoriesAllowList := getEnvWithDefaultEmptySet("DELTA_BUILD_REPOS_ALLOWLIST")
+	if len(deltaBuildRepositoriesAllowList) > 0 {
+		var repos []string
+		for r := range deltaBuildRepositoriesAllowList {
+			repos = append(repos, r)
 		}
 
-		debug.Printf("capturing separate indexing metrics for: %s", repos)
+		debug.Printf("using delta shard builds for: %s", strings.Join(repos, ", "))
```

Comment:
minor: can you add a tiny helper to build string for a set. Then both if conditionals collapse into just `debug.Printf(..., setValues(deltaBuildRepositoriesAllowList)`.